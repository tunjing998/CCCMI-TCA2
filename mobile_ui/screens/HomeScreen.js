import React from 'react';
import {View, Text, StyleSheet, StatusBar} from 'react-native';
import {Button} from 'react-native-elements';
import {useTheme} from '@react-navigation/native';
import LinearGradient from 'react-native-linear-gradient';
import testVariables from '../appium_automation_testing/test_variables';

const HomeScreen = ({navigation}) => {
  const {colors} = useTheme();

  const theme = useTheme();

  return (
    <View
      style={styles.container}
      accessibilityLabel={testVariables.homeScreenContainer}
      testID={'homeScreenContainer'}>
      <StatusBar barStyle={theme.dark ? 'light-content' : 'dark-content'} />

      <Button
        testID={'homeScreenTakeNewSampleButton'}
        ViewComponent={LinearGradient}
        linearGradientProps={{
          colors: ['#e76f51', '#f4a261'],
          start: {x: 0, y: 0},
          end: {x: 1, y: 0},
        }}
        title="Take new sample"
        titleStyle={styles.title}
        containerStyle={styles.buttonContainer}
        onPress={() => navigation.navigate('SearchRiverScreen')}
        accessibilityLabel={testVariables.homeScreenTakeNewSampleButton}
      />
      <Button
        accessibilityLabel={testVariables.homeScreenViewSampleButton}
        testID={testVariables.homeScreenViewSampleButton}
        ViewComponent={LinearGradient}
        linearGradientProps={{
          colors: ['#e76f51', '#f4a261'],
          start: {x: 0, y: 0},
          end: {x: 1, y: 0},
        }}
        title="View sample"
        titleStyle={styles.title}
        containerStyle={styles.buttonContainer}
      />
      <Button
        title="upload image test"
        onPress={() => navigation.navigate('uploadImage')}
      />
      <Button
        title="upload image to server test"
        onPress={() => navigation.navigate('uploadImageToServer')}
      />
      <Button
        title="result test"
        onPress={() => navigation.navigate('resultPage')}
      />
    </View>
  );
};

export default HomeScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    fontSize: 31,
    paddingHorizontal: 30,
    paddingVertical: 10,
  },
  buttonContainer: {
    padding: 35,
  },
});
