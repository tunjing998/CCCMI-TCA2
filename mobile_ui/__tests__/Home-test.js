import React, {Component} from 'react';
import {View, Text} from 'react-native';
import HomeScreen from '../screens/HomeScreen';
import renderer from 'react-test-renderer';

test('Home screen snapShot', () => {
  const snap = renderer.create(<HomeScreen />).toJSON();
  expect(snap).toMatchSnapshot();
});
