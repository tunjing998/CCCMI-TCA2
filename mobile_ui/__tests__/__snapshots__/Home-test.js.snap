// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen snapShot 1`] = `
<View
  accessibilityLabel="homeScreenContainer"
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="homeScreenContainer"
>
  <View
    style={
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
        "padding": 35,
      }
    }
  >
    <View
      accessibilityLabel="homeScreenTakeNewSampleButton"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="homeScreenTakeNewSampleButton"
    >
      <BVLinearGradient
        colors={
          Array [
            4293357393,
            4294222433,
          ]
        }
        endPoint={
          Object {
            "x": 1,
            "y": 0,
          }
        }
        locations={null}
        startPoint={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 31,
              "paddingHorizontal": 30,
              "paddingVertical": 10,
              "textAlign": "center",
            }
          }
        >
          Take new sample
        </Text>
      </BVLinearGradient>
    </View>
  </View>
  <View
    style={
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
        "padding": 35,
      }
    }
  >
    <View
      accessibilityLabel="homeScreenViewSampleButton"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="homeScreenViewSampleButton"
    >
      <BVLinearGradient
        colors={
          Array [
            4293357393,
            4294222433,
          ]
        }
        endPoint={
          Object {
            "x": 1,
            "y": 0,
          }
        }
        locations={null}
        startPoint={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 31,
              "paddingHorizontal": 30,
              "paddingVertical": 10,
              "textAlign": "center",
            }
          }
        >
          View sample
        </Text>
      </BVLinearGradient>
    </View>
  </View>
  <View
    style={
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          upload image test
        </Text>
      </View>
    </View>
  </View>
</View>
`;
